src/interpreter/bytecode-generator.cc

```diff
@@ -5647,8 +5647,34 @@ void BytecodeGenerator::VisitCall(Call* expr) {
 
   builder()->SetExpressionPosition(expr);
 
-  // Emit a breakpoint for all call expressions.
-  builder()->RecordReplayInstrumentation("breakpoint", expr->position());
+  // Emit a breakpoint for all call expressions
+  // after arguments have already been evaluated.
+  // This might duplicate the call's parent's position,
+  // so we should try to have it get deduplicated
+  // by inserting it before the call, *if* there are no arguments.
+  //
+  // Example1 (potentially deduped breakpoint):
+  // `/*BREAK1*/func();
+  //
+  // Example2 (extra breakpoint after arguments evaluation):
+  // `/*BREAK1*/func/*BREAK3*/(/*BREAK2*/g());`
+
+  // TODO: Don't emit a duplicate breakpoint if there are no nested calls.
+  //       Example: `/*BREAK1*/func/*BREAK2*/(a, b);`
```

Comment:
based on my testing it would really be great if we could fix this before landing this patch